### PR TITLE
Added attachment option for slash commands

### PR DESCRIPTION
**Pull request by Galarzaa90**
I don't know if there were any plans to implement this differently, but I decided to try it out.

The only problem is that this converter does not work with chat commands, and it will always fail if used in a chat command. I don't know if I should do something about it.

image

### CODE REVIEW

File: kord-extensions/src/main/kotlin/com/kotlindiscord/kord/extensions/commands/converters/impl/AttachmentConverter.kt [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    "attachment",
+
+    types = [ConverterType.OPTIONAL, ConverterType.SINGLE],
+)
+public class AttachmentConverter(
+    override var validator: Validator<Attachment> = null
+) : SingleConverter<Attachment>() {
+    override val signatureTypeString: String = "converters.attachment.signatureType"
+
+    override suspend fun parse(parser: StringParser?, context: CommandContext, named: String?): Boolean = false
```
Comment (gdude2002) [inline kord-extensions/src/main/kotlin/com/kotlindiscord/kord/extensions/commands/converters/impl/AttachmentConverter.kt]: This needs a more descriptive, translated error - thrown in a `DiscordRelayedException`. I think there's already a translation string you can use in there, but I'm not sure.

File: kord-extensions/src/main/resources/translations/kordex/strings.properties [hunk 2]
```diff
@@ -68,6 +68,7 @@ commands.error.missingBotPermissions=I don't have the permissions I need to run
 commands.error.user=Unfortunately, **an error occurred** during command processing. Please let a staff member know.
 commands.error.user.sentry.message=Unfortunately, **an error occurred** during command processing. If you'd like to submit information on what you were doing when this error happened, please use the following command\: ```{0}feedback {1} <message>```
 commands.error.user.sentry.slash=Unfortunately, **an error occurred** during command processing. If you'd like to submit information on what you were doing when this error happened, please use the following command\: ```/feedback {0} <message>```
+converters.attachment.signatureType=attachment
```
Comment (gdude2002) [inline kord-extensions/src/main/resources/translations/kordex/strings.properties:71]: This needs to be added to the English translations file as well.

**Review by gdude2002 — CHANGES_REQUESTED** [review]

File: kord-extensions/src/main/resources/translations/kordex/strings.properties (hunk 2 above)
Comment (Galarzaa90) [reply kord-extensions/src/main/resources/translations/kordex/strings.properties:71]: Done, added to `strings_en_GB.properties`

File: kord-extensions/src/main/kotlin/com/kotlindiscord/kord/extensions/commands/converters/impl/AttachmentConverter.kt (hunk 1 above)
Comment (Galarzaa90) [reply kord-extensions/src/main/kotlin/com/kotlindiscord/kord/extensions/commands/converters/impl/AttachmentConverter.kt]: I added a DiscordRelayedException, didn't find a translation string for it, added a new one.

I found a very similar case, for `ChoiceConverters`, when trying to use one in a chat command, it throws an IllegalStateException:
```kt
is ChoiceConverter<*> -> error("Choice converters may only be used with slash commands")
```
https://github.com/Kord-Extensions/kord-extensions/blob/develop/kord-extensions/src/main/kotlin/com/kotlindiscord/kord/extensions/commands/chat/ChatCommandParser.kt#L119

Comment (gdude2002) [reply kord-extensions/src/main/kotlin/com/kotlindiscord/kord/extensions/commands/converters/impl/AttachmentConverter.kt]: That's in the parser though, which isn't entirely ideal, yeah. I'll probably change how that works someday.

**Comment by gdude2002** [comment]
Looking good - thanks!

**Pull request merged**